apps/lib/apps_ui.c

```diff
@@ -37,7 +37,7 @@ static int ui_read(UI *ui, UI_STRING *uis)
             {
                 const char *password =
                     ((PW_CB_DATA *)UI_get0_user_data(ui))->password;
-                if (password && password[0] != '\0') {
+                if (password != NULL) {
```

Comment:
Please add an empty line after the var decl.

Reply:
While you are at it: 
there are three cases in this source file where a pointer test is written without `!= NULL`,
which may be improved along with the other improvements: `if (opener)` above and `if (reader)` and `if (closer)` below.